### DISCUSSION TITLE
App Router - How to Fetch and Maintain global data for all components

### DISCUSSION THREAD

**Discussion by piotrkulpinski** [vercel/next.js#48535] [Help]
### Summary

So I'm trying the new Next 13 app router and I'm trying to build a multi-tenant application, following [this guide](https://vercel.com/guides/nextjs-multi-tenant-application) and trying to port it to the app directory.

I'm using middleware.ts to grab the subdomain and rewrite the url to `app/_sites/[siteSlug]/`. 
In the root layout of this file, I grab the params.siteSlug and fetch the Site from the database. 

Everything works fine, except that I'd like to pass the site object to other server side components that are fetching other info related to that site. 

I could pass it down as props, but as the app grows, that would be hard to maintain.

So the question is, is it possible to store the fetched data in a global state that is **accessible from the server side**? Normally I'd use `useContext` but it won't work with the server side components. 

Any advice would be appreciated.

**Comment by imNotTheDevYouAreLookingFor** [comment]
How did you end up solving this?

**Reply by piotrkulpinski** [reply]
I've moved to Remix.run for this project, but one solution I think would be to store the data in a server-side compatible state manager like https://github.com/pmndrs/zustand.
